File: apps/dashboard/src/types/risk.ts
export enum RiskLevel {
  VERY_LOW = "VERY_LOW",
  LOW = "LOW",
  MEDIUM = "MEDIUM",
  HIGH = "HIGH",
  VERY_HIGH = "VERY_HIGH",
  CRITICAL = "CRITICAL",
}

export enum RiskVerdict {
  APPROVE = "APPROVE",
  REVIEW = "REVIEW",
  DECLINE = "DECLINE",
  BLOCK = "BLOCK",
}

export interface RiskScore {
  score: number;
  level: RiskLevel;
  verdict: RiskVerdict;
  confidence: number;
  model_version: string;
  scored_at: string;
}

export interface FeatureImportance {
  feature_name: string;
  value: number | string;
  importance: number;
  direction: "increases_risk" | "decreases_risk";
}

export interface RiskExplanation {
  risk_score: RiskScore;
  top_features: FeatureImportance[];
  shap_values: Record<string, number>;
  graph_evidence: Array<Record<string, unknown>>;
  behavioral_deviations: string[];
  counterfactual: string;
  natural_language_explanation: string;
  similar_cases: Array<Record<string, unknown>>;
  rules_triggered: string[];
}

export interface RiskThresholds {
  high: number;
  very_high: number;
  critical: number;
  auto_block: number;
}

export interface RiskHeatmapCell {
  hour: number;
  day: number;
  value: number;
}

File: core/schemas/risk.py
from enum import Enum
from typing import List, Dict, Any, Optional
from datetime import datetime
from pydantic import Field
from .base import BaseSchema


class RiskLevel(str, Enum):
    VERY_LOW = "VERY_LOW"
    LOW = "LOW"
    MEDIUM = "MEDIUM"
    HIGH = "HIGH"
    VERY_HIGH = "VERY_HIGH"
    CRITICAL = "CRITICAL"


class RiskVerdict(str, Enum):
    APPROVE = "APPROVE"
    REVIEW = "REVIEW"
    DECLINE = "DECLINE"
    BLOCK = "BLOCK"


class RiskScore(BaseSchema):
    score: float = Field(ge=0.0, le=1.0)
    level: RiskLevel
    verdict: RiskVerdict
    confidence: float = Field(ge=0.0, le=1.0)
    model_version: str
    scored_at: datetime = Field(default_factory=datetime.utcnow)


class FeatureImportance(BaseSchema):
    feature_name: str
    value: Any
    importance: float
    direction: str  # e.g., "increases_risk", "decreases_risk"


class RiskExplanation(BaseSchema):
    risk_score: RiskScore
    top_features: List[FeatureImportance]
    shap_values: Dict[str, float]
    graph_evidence: List[Dict[str, Any]]
    behavioral_deviations: List[str]
    counterfactual: str
    natural_language_explanation: str
    similar_cases: List[Dict[str, Any]]
    rules_triggered: List[str]
